### PR TITLE
Fix service debbug hit issue

### PR DESCRIPTION
**Pull request by praveennadarajah**
## Purpose
This PR will fix service debug hit issue when a service file is located inside a directory.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/23096

### CODE REVIEW

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java [hunk 1]
```diff
@@ -86,4 +89,24 @@ public static boolean isBlank(String str) {
...
+    public static String getDirectoryRelativePath(String sourcePath, String patternMatcher, String sourceName,
+                                                  String fileName) {
+        Pattern pattern = Pattern.compile(SERVICE_REGEX);
+        Matcher matcher = pattern.matcher(patternMatcher);
+        if (matcher.find()) {
+            return sourcePath;
+        } else {
+            return sourcePath.replace(sourceName, fileName);
+        }
+    }
```
Comment (NipunaRanasinghe) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java:111]: @praveennadarajah Shall we add a comment explaining that in here we have handled the service files specifically and the reason behind that?

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java [hunk 2]
```diff
@@ -86,4 +89,24 @@ public static boolean isBlank(String str) {
...
+    /**
+     * Get relative path when a bal file is inside a directory.
+     *
+     * @param sourcePath     source path
+     * @param patternMatcher pattern matcher string
+     * @param sourceName     source name
+     * @param fileName       file name
+     * @return relative path
+     */
+    public static String getDirectoryRelativePath(String sourcePath, String patternMatcher, String sourceName,
```
Comment (NipunaRanasinghe) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java:102]: What do you mean by `patternMatcher` here? AFAIR this the reference location of the source that we are rectifying, right? we can think of using an appropriate arg name instead.

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java (hunk 1 above)
Comment (praveennadarajah) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java]: Added comments.

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java (hunk 2 above)
Comment (praveennadarajah) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java:102]: Yes, patternMatcher refers to "reference location" in one instance and "reference type" in another instance. I think renaming patternMatcher to "**reference**" would work fine. WDYT?

Comment (NipunaRanasinghe) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java:102]: +1 for `reference`/`referencePath`

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java (hunk 1 above)
Comment (NipunaRanasinghe) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java]: Looks good!

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java (hunk 2 above)
Comment (praveennadarajah) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java]: Renamed to `reference`.

**Pull request merged**